### ISSUE TITLE
Mongoskin error

### ISSUE THREAD

**Issue by jgamedev** [Nimos/joterblox#1]
1) cloned the repo
2) npm install
3) service mongod start
4) node Game.js

Got this error:

> TypeError: Cannot read property 'name' of undefined
>     at makeSkinClass (/home/jay/node_projects/joterblox/node_modules/mongoskin/lib/utils.js:33:43)
>     at Object.<anonymous> (/home/jay/node_projects/joterblox/node_modules/mongoskin/lib/grid.js:6:35)
>     at Module._compile (module.js:541:32)
>     at Object.Module._extensions..js (module.js:550:10)
>     at Module.load (module.js:458:32)
>     at tryModuleLoad (module.js:417:12)
>     at Function.Module._load (module.js:409:3)
>     at Module.require (module.js:468:17)
>     at require (internal/module.js:20:19)
>     at Object.<anonymous> (/home/jay/node_projects/joterblox/node_modules/mongoskin/lib/db.js:22:16)

**Comment by mosni** [comment]
Apparently NPM changed the way it handles peer dependencies in a recent version. The pushed package.json should fix it. Pull, npm install, try again and report back please.

**Comment by jgamedev** [comment]
We have made progress!! Thank you.

I can get to the start start screen and se a blinking press any key button.
But that's as far as I get because appearently the server crashs. Here's the log:

>  joao@joao-13Z940-G-BK71P1  ~/node_projects/joterblox   master  node Game.js
> { Error: Cannot find module '../build/Release/bson'
>     at Function.Module._resolveFilename (module.js:440:15)
>     at Function.Module._load (module.js:388:25)
>     at Module.require (module.js:468:17)
>     at require (internal/module.js:20:19)
>     at Object.<anonymous> (/home/joao/node_projects/joterblox/node_modules/bson/ext/index.js:15:10)
>     at Module._compile (module.js:541:32)
>     at Object.Module._extensions..js (module.js:550:10)
>     at Module.load (module.js:458:32)
>     at tryModuleLoad (module.js:417:12)
>     at Function.Module._load (module.js:409:3) code: 'MODULE_NOT_FOUND' }
> js-bson: Failed to load c++ bson extension, using pure JS version
> [Game] Starting a new round
> listening on *:3000
> [Info] New player
> undefined
> /home/joao/node_projects/joterblox/node_modules/mongodb/lib/mongodb/mongo_client.js:409
>           throw err
>           ^

TypeError: Cannot read property '0' of undefined
    at Promise.<anonymous> (/home/jay/node_projects/joterblox/classes/Profile.js:76:14)
    at Promise.<anonymous> (/home/jay/node_projects/joterblox/node_modules/mpromise/lib/promise.js:173:8)
    at emitOne (events.js:101:20)
    at Promise.emit (events.js:188:7)
    at Promise.emit (/home/jay/node_projects/joterblox/node_modules/mpromise/lib/promise.js:84:38)
    at Promise.reject (/home/jay/node_projects/joterblox/node_modules/mpromise/lib/promise.js:112:15)
    at Promise.resolve (/home/jay/node_projects/joterblox/node_modules/mpromise/lib/promise.js:125:24)
    at onError (/home/jay/node_projects/joterblox/node_modules/mongoskin/lib/utils.js:27:7)
    at EventEmitter.<anonymous> (/home/jay/node_projects/joterblox/node_modules/mongoskin/lib/utils.js:118:15)
    at EventEmitter.g (events.js:286:16)

**Comment by mosni** [comment]
that looks like your mongodb isn't running, or isn't setup correctly. You can change the database settings in settings.js, although the default setting there should work with the mongod default port.

**Comment by jgamedev** [comment]
Exactly. It works now and turns out it's freaking awesome!

3 more things:

1) how many players do you think it supports on the same map before frame rates go down noticeably?

2) Are there any known bugs?

3) I think you are missing a license for the codebase? If so, please consider adding it.

Thanks,
- Jay

**Comment by jgamedev** [comment]
Sorry.. I forgot the most important question:

4)  Is  the server authoritative?  Or is it just passing along whatever data is submited by the clients?

**Comment by mosni** [comment]
1) I think we tested with up to 7, without issues
2) I don't think any major known bugs are left. We abandoned the game because we lost interest, but from a quick glance, everything on our own issue tracker are features or UI tweaks.
3) I added an AGPL license, which I think is appropriate.
4) All game logic is calculated server side, the clients just send back inputs. This isn't optimal, as network lag becomes very apparent with higher latencies, but for the "normal" ~30ms ping of people living within a few hundred kilometers of the server, it's barely noticable.
